From 34d8c9d5b4e4fc62bea212b6bd046f26a177df4c Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Mon, 7 Aug 2023 10:23:16 -0700
Subject: [PATCH 158/180] ld: Build libpr23169a.so with -z lazy

pr23169b test only works with lazy binding.  To work with linker which
disables lazy binding by default, build pr23169b binaries with -z lazy.

	PR ld/30698
	* ld-ifunc/ifunc.exp: Build pr23169b binaries with -z lazy.

(cherry picked from commit 51dd9e7c4cab13748f31f6575aed0672f03d6710)
---
 ld/testsuite/ld-ifunc/ifunc.exp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ld/testsuite/ld-ifunc/ifunc.exp b/ld/testsuite/ld-ifunc/ifunc.exp
index edabcc3f744..2bf656e73b2 100644
--- a/ld/testsuite/ld-ifunc/ifunc.exp
+++ b/ld/testsuite/ld-ifunc/ifunc.exp
@@ -604,7 +604,7 @@ run_cc_link_tests [list \
     ] \
     [list \
 	"Build libpr23169a.so" \
-	"-shared" \
+	"-shared -Wl,-z,lazy" \
 	"-fPIC -O2 -g" \
 	{ pr23169a.c } \
 	{} \
@@ -751,7 +751,7 @@ run_ld_link_exec_tests [list \
     ] \
     [list \
 	"Run pr23169b" \
-	"-pie -Wl,--no-as-needed tmpdir/libpr23169a.so" \
+	"-pie -Wl,--no-as-needed,-z,lazy tmpdir/libpr23169a.so" \
 	"" \
 	{ pr23169b.c pr23169c.c } \
 	"pr23169b" \
-- 
2.43.0

